### ISSUE TITLE
404 Handling is too aggressive

### ISSUE THREAD

**Issue by burkel24** [rcos/Observatory3#232]
Right now every api response that 404s redirects the user to the homepage. There are many valid reasons an api call might 404 that shouldn't result in a redirect.

**Comment by seveibar** [comment]
@burkel24 @agundy What should happen? And what are these bad endpoints?

**Comment by blivingston24** [comment]
@seveibar The first time I hit this problem was when I initially ran the app w/o populating the database. The call for class years would fail on many pages and made it possible to create an acct or login.

Another case was adding a project. I entered an invalid url for the github link and the fetch failed, leading to the redirect once again. 

The former case I could see as being acceptable behavior (though not ideal as the user would have no idea what is happening) but the latter certainly needs to be handled better.

**Comment by agundy** [comment]
I agree with Burke, I think we should look into handling errors better for adding forms and what not. We can start with his examples but look for other spots too.
